director/director_advertise_test.go

```diff
@@ -22,21 +22,108 @@ package director_test
 
 import (
 	"encoding/json"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"sync/atomic"
 	"testing"
+	"time"
 
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/pelicanplatform/pelican/fed_test_utils"
+	"github.com/pelicanplatform/pelican/param"
 	"github.com/pelicanplatform/pelican/server_structs"
 	"github.com/pelicanplatform/pelican/server_utils"
 )
 
+// Test for a director disappearing
+func TestDirectorShutdown(t *testing.T) {
+	server_utils.ResetTestState()
+
```

Comment:
Can you add a defer statement to also reset test state? I think it's better to have tests clean up after themselves in addition to starting with a clean state.